### PR TITLE
copyTransient is not global

### PR DESCRIPTION
**Pull request by prasanthj**
Fix for #370 

There were couple of approaches discussed in mailing list. Making it a config in kryo object seems to be easier.

### CODE REVIEW

**Comment by romix** [comment]
I understand that you want to have a global setting for copyTransient inside a Kryo instance. But I'm not quite sure about removing the ability to set it on per FieldSerializer basis. I think it should still be possible to change it for a concrete instance of FieldSerializer. Instead, in the constructor of FieldSerializer you could take the initial value for copyTransient from the Kryo instance. Does it make sense to you?

**Comment by prasanthj** [comment]
Yes. @romix That makes sense. I will make those changes and will update the pull request so as to not break compatibility.

**Comment by prasanthj** [comment]
@romix Updated the PR as per your review comments.

**Comment by prasanthj** [comment]
I will a test case shortly

**Comment by prasanthj** [comment]
@romix Can you please review this PR?

File: test/com/esotericsoftware/kryo/FieldSerializerTest.java [hunk 1]
```diff
@@ -15,8 +15,8 @@
  * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
  * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
  * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
- * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
-
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
+
```
Comment (romix) [inline test/com/esotericsoftware/kryo/FieldSerializerTest.java]: What is this file header touched at all?

File: src/com/esotericsoftware/kryo/Kryo.java [hunk 2]
```diff
@@ -1046,6 +1047,24 @@ public void setCopyReferences (boolean copyReferences) {
 		this.copyReferences = copyReferences;
 	}
 
+	/**
+	 * If false, when {@link #copy(Object)} is called all transient fields that are accessible will be ignored from
+	 * being copied. This has to be set before registering classes with kryo for it to be used by all field
+	 * serializers. If transient fields has to be copied for specific classes then use {@link FieldSerializer#setCopyTransient(boolean)}.
+	 * Default is true.
+   */
```
Comment (romix) [inline src/com/esotericsoftware/kryo/Kryo.java]: Problem with indentation?

**Comment by romix** [comment]
Overall, the PR looks good to me.

I've commented on a few indentation issues. May be there are others. Please have a look and fix them.

@NathanSweet @magro I think with those small formatting issues fixed, it is OK to merge. Any objections?

File: test/com/esotericsoftware/kryo/FieldSerializerTest.java (hunk 1 above)
Comment (prasanthj) [inline test/com/esotericsoftware/kryo/FieldSerializerTest.java]: Line no: 18, 19 and EOF was missing CRLF. IntelliJ inserted the line feed automatically.

File: src/com/esotericsoftware/kryo/Kryo.java (hunk 2 above)
Comment (prasanthj) [inline src/com/esotericsoftware/kryo/Kryo.java]: Fixed it in the latest commit.

**Comment by prasanthj** [comment]
Sorry to bother again. We are looking forward for this feature in Apache Hive. It will be great if we can get this in.

**Comment by magro** [comment]
LGTM. @NathanSweet?

**Pull request merged**